creepy/subprocess/pypen.py

```diff
@@ -108,4 +125,9 @@ def detach(self):
 except Exception:
     import traceback
     traceback.print_exc()
+finally:
+    try:
+        os.write({fdw}, b'\\0')
```

Comment:
What's the point of writing characters \ and 0?

Reply:
It writes single \0 character to break `recv()` on the other end. Otherwise the other end is waiting for data even if file descriptor is closed.

Reply:
```
import os

fd = os.open("test.txt", os.O_RDWR|os.O_CREAT)
os.write(fd, b'\\0')
os.close(fd)


with open('test.txt') as file:
    for line in file.readlines():
        for c in line:
            print(c)
```

Prints
```
\
0
```

Reply:
Wrap this code between triple quotes """ and exec.